code/game/objects/structures/plasticflaps.dm

```diff
@@ -17,6 +17,17 @@
 	. = ..()
 	alpha = 0
 	gen_overlay()
+	update_atmos_behaviour()
+
+/obj/structure/plasticflaps/Destroy()
+	update_atmos_behaviour()
+	return ..()
+
+/obj/structure/plasticflaps/Move(atom/newloc, direct, glide_size_override)
+	var/turf/old_loc = loc
+	. = ..()
+	update_atmos_behaviour()
```

Comment:
why are you calling update_atmos_behavior here? seems wrong

Reply:
i've put the call in Moved()